The unit tests are broken: 

```
error[E0425]: unresolved name `PositionContainer`
   --> /home/travis/build/servo/servo/tests/unit/style/properties/serialization.rs:698:49
    |
698 |             let position = DeclaredValue::Value(PositionContainer(
    |                                                 ^^^^^^^^^^^^^^^^^
error[E0425]: unresolved name `PositionContainer`
   --> /home/travis/build/servo/servo/tests/unit/style/properties/serialization.rs:749:49
    |
749 |             let position = DeclaredValue::Value(PositionContainer(
    |                                                 ^^^^^^^^^^^^^^^^^
error[E0425]: unresolved name `PositionContainer`
   --> /home/travis/build/servo/servo/tests/unit/style/properties/serialization.rs:799:49
    |
799 |             let position = DeclaredValue::Value(PositionContainer(
    |                                                 ^^^^^^^^^^^^^^^^^
```

--- a/components/layout/display_list_builder.rs
+++ b/components/layout/display_list_builder.rs
@@ -520,9 +520,9 @@ impl FragmentDisplayListBuilding for Fragment {
             };
 
             // Use `background-position` to get the offset.
-            let horizontal_position = model::specified(background.background_position.0.horizontal,
+            let horizontal_position = model::specified(background.background_position.horizontal,
                                                        bounds.size.width - image_size.width);
-            let vertical_position = model::specified(background.background_position.0.vertical,
+            let vertical_position = model::specified(background.background_position.vertical,
                                                      bounds.size.height - image_size.height);
 
             let abs_x = border.left + virtual_origin_x + horizontal_position + origin_x;

--- a/components/style/properties/gecko.mako.rs
+++ b/components/style/properties/gecko.mako.rs
@@ -1041,16 +1041,16 @@ fn static_assert() {
     pub fn clone_background_position(&self) -> longhands::background_position::computed_value::T {
         use values::computed::position::Position;
         let position = &self.gecko.mImage.mLayers.mFirstElement.mPosition;
-        longhands::background_position::computed_value::T(Position {
+        Position {
             horizontal: position.mXPosition.into(),
             vertical: position.mYPosition.into(),
-        })
+        }
     }
 
     pub fn set_background_position(&mut self, v: longhands::background_position::computed_value::T) {
         let position = &mut self.gecko.mImage.mLayers.mFirstElement.mPosition;
-        position.mXPosition = v.0.horizontal.into();
-        position.mYPosition = v.0.vertical.into();
+        position.mXPosition = v.horizontal.into();
+        position.mYPosition = v.vertical.into();
         self.gecko.mImage.mPositionXCount = 1;
         self.gecko.mImage.mPositionYCount = 1;
     }

--- a/components/style/properties/helpers/animated_properties.mako.rs
+++ b/components/style/properties/helpers/animated_properties.mako.rs
@@ -496,13 +496,6 @@ impl Interpolate for Position {
     }
 }
 
-impl Interpolate for BackgroundPosition {
-    #[inline]
-    fn interpolate(&self, other: &Self, time: f64) -> Result<Self, ()> {
-        Ok(BackgroundPosition(try!(self.0.interpolate(&other.0, time))))
-    }
-}
-
 impl Interpolate for BackgroundSize {
     fn interpolate(&self, other: &Self, time: f64) -> Result<Self, ()> {
         use properties::longhands::background_size::computed_value::ExplicitSize;

--- a/components/style/properties/longhand/background.mako.rs
+++ b/components/style/properties/longhand/background.mako.rs
@@ -85,54 +85,23 @@ ${helpers.predefined_type("background-color", "CSSColor",
         pub mod computed_value {
             use values::computed::position::Position;
 
-            #[derive(PartialEq, Copy, Clone, Debug)]
-            #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
-            pub struct T(pub Position);
+            pub type T = Position;
         }
 
-        impl HasViewportPercentage for SpecifiedValue {
-            fn has_viewport_percentage(&self) -> bool {
-                self.0.has_viewport_percentage()
-            }
-        }
-
-        #[derive(Debug, Clone, PartialEq, Copy)]
-        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
-        pub struct SpecifiedValue(pub Position);
-
-        impl ToCss for SpecifiedValue {
-            fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-                self.0.to_css(dest)
-            }
-        }
-
-        impl ToCss for computed_value::T {
-            fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-                self.0.to_css(dest)
-            }
-        }
-
-        impl ToComputedValue for SpecifiedValue {
-            type ComputedValue = computed_value::T;
-
-            #[inline]
-            fn to_computed_value(&self, context: &Context) -> computed_value::T {
-                computed_value::T(self.0.to_computed_value(context))
-            }
-        }
+        pub type SpecifiedValue = Position;
 
         #[inline]
         pub fn get_initial_value() -> computed_value::T {
             use values::computed::position::Position;
-            computed_value::T(Position {
+            Position {
                 horizontal: computed::LengthOrPercentage::Percentage(0.0),
                 vertical: computed::LengthOrPercentage::Percentage(0.0),
-            })
+            }
         }
 
         pub fn parse(_context: &ParserContext, input: &mut Parser)
                      -> Result<SpecifiedValue, ()> {
-            Ok(SpecifiedValue(try!(Position::parse(input))))
+            Ok(try!(Position::parse(input)))
         }
 </%helpers:longhand>
 

--- a/tests/unit/style/properties/serialization.rs
+++ b/tests/unit/style/properties/serialization.rs
@@ -695,12 +695,12 @@ mod shorthand_serialization {
                 authored: None
             });
 
-            let position = DeclaredValue::Value(PositionContainer(
+            let position = DeclaredValue::Value(
                 Position {
                     horizontal: LengthOrPercentage::Length(Length::from_px(7f32)),
                     vertical: LengthOrPercentage::Length(Length::from_px(4f32))
                 }
-            ));
+            );
 
             let repeat = DeclaredValue::Value(Repeat::repeat_x);
             let attachment = DeclaredValue::Value(Attachment::scroll);
@@ -746,12 +746,12 @@ mod shorthand_serialization {
                 authored: None
             });
 
-            let position = DeclaredValue::Value(PositionContainer(
+            let position = DeclaredValue::Value(
                 Position {
                     horizontal: LengthOrPercentage::Length(Length::from_px(7f32)),
                     vertical: LengthOrPercentage::Length(Length::from_px(4f32))
                 }
-            ));
+            );
 
             let repeat = DeclaredValue::Value(Repeat::repeat_x);
             let attachment = DeclaredValue::Value(Attachment::scroll);
@@ -796,12 +796,12 @@ mod shorthand_serialization {
                 authored: None
             });
 
-            let position = DeclaredValue::Value(PositionContainer(
+            let position = DeclaredValue::Value(
                 Position {
                     horizontal: LengthOrPercentage::Length(Length::from_px(0f32)),
                     vertical: LengthOrPercentage::Length(Length::from_px(0f32))
                 }
-            ));
+            );
 
             let repeat = DeclaredValue::Value(Repeat::repeat_x);
             let attachment = DeclaredValue::Value(Attachment::scroll);

